uc-cdis/sheepdog
# PXP-4102 Feat/arb admin

Switches Sheepdog authz handling of program/project CRUD to centralized auth/Arborist.

See https://github.com/uc-cdis/commons-users/pull/810 for new Sheepdog admin policy.

(For QAs' benefit) How to test:
- See PR linked above for example of how to toggle for a given user the old `admin: true/false` and the new admin policy 
- Verify that the API behaves according to this matrix:
```
    User DOES have Sheepdog admin policy: All CRUD authorized.
        Admin is FALSE:
            Create program: Authorized
            Delete program: Authorized
            Create project: Authorized
            Delete project: Authorized
        Admin is TRUE:
            Create program: Authorized
            Delete program: Authorized
            Create project: Authorized
            Delete project: Authorized
    User does NOT have Sheepdog admin policy: All CRUD not authorized.
    (Also check that they are 403s in particular)
        Admin is FALSE:
            Create program: Unauthorized
            Delete program: Unauthorized
            Create project: Unauthorized
            Delete project: Unauthorized
        Admin is TRUE:
            Create program: Unauthorized
            Delete program: Unauthorized
            Create project: Unauthorized
            Delete project: Unauthorized
```

### New Features
Switch checks guarding program/project CRUD to centralized auth (instead of checking for "admin" in JWT, check for Sheepdog admin policy in Arborist)

### Breaking Changes
Having admin: true in user.yaml no longer allows a user to do program/project CRUD in Sheepdog. Users now need to have the Sheepdog admin policy in Arborist. 

### Deployment changes
The environment's user.yaml needs to be updated to add the new Sheepdog admin policy/resources/roles; for each user that has admin: true, grant them the new policy. See https://github.com/uc-cdis/commons-users/pull/810 for an example.
